src/Missings.jl

```diff
@@ -1,51 +1,216 @@
 __precompile__(true)
 module Missings
 
-import Base: *, <, ==, !=, <=, !, +, -, ^, /, &, |, xor
 using Compat
 
 export allowmissing, disallowmissing, ismissing, missing, missings,
        Missing, MissingException, levels, skipmissing
 
-"""
-    Missing
+if VERSION < v"0.7.0-DEV.2762"        
```

Comment:
It's kind of weird, but if I define the types in the same `if` block as the rest I get ` type definition not allowed inside a local scope` errors.

Reply:
That's really bizarre. What if you put an `@static` annotation on the `if`?

Reply:
Same. AFAIK `@static` has no effect at the top level. I've found this issue, but it's pretty old and apparently closed: https://github.com/JuliaLang/julia/issues/10555

Reply:
`@static` is used at the top level in Compat.jl: https://github.com/JuliaLang/Compat.jl/blob/master/src/Compat.jl

Reply:
Yes, but I can swear that I've read a comment by Jameson saying that has no effect.

Reply:
I used it to hide old deprecated syntax from 0.7: https://github.com/JuliaLang/Compat.jl/pull/419/files

Reply:
My understanding is that since the condition is evaluated at parse time, it gives an opportunity to trick the parser into ignoring things doesn't particularly like.